R/hash.R

```diff
@@ -3,6 +3,44 @@ renv_hash_text <- function(text) {
   renv_bootstrap_hash_text(text)
 }
 
+renv_hash_fields <- function(dcf) {
+  c(
+    renv_hash_fields_default(),
+    renv_hash_fields_remotes(dcf)
+  )
+}
+
+renv_hash_fields_default <- function() {
+  c(
+    "Package", "Version", "Title",
```

Comment:
Title, Authors, Maintainer, and Description are not necessary for hashing, as they do not communicate anything about package dependencies.

What about SystemRequirements and NeedsCompilation?

Reply:
FWIW if we change the fields that get included in the hash, then we'll invalidate any old hash values. We could still do this, but we'd probably want to do something to either notify users or allow them to choose.

Ultimately though, we want to include fields which would help to uniquely identify a package from its DESCRIPTION file, hence why those (and not just the package dependency fields) are included.